src/parser/view_transform.c

```diff
@@ -3610,6 +3610,113 @@ mq_rewrite_vclass_spec_as_derived (PARSER_CONTEXT * parser, PT_NODE * statement,
   return spec;
 }
 
+/*
+ * mq_rewrite_dblink_as_derived -
+ *   return: rewritten dblink as a derived table (like a subquery)
+ *   parser(in): 
+ *   query(in): it should be a spec node with dblink
+ */
+static PT_NODE *
+mq_rewrite_dblink_as_derived (PARSER_CONTEXT * parser, PT_NODE * query)
+{
+  PT_NODE *new_query = NULL, *derived = NULL;
+  PT_NODE *range = NULL, *spec = NULL, *temp, *node = NULL;
+  PT_NODE **head;
+  int i = 0;
+
+  /* set line number to range name */
+  range = pt_name (parser, "_dbl");
+  if (range == NULL)
+    {
+      PT_INTERNAL_ERROR (parser, "allocate new node");
+      goto exit_on_error;
+    }
+
+  /* construct new spec We are now copying the query and updating the spec_id references */
+  spec = parser_new_node (parser, PT_SPEC);
+  if (spec == NULL)
+    {
+      PT_INTERNAL_ERROR (parser, "allocate new node");
+      goto exit_on_error;
+    }
+
+  derived = query;
+  derived = mq_reset_ids_in_statement (parser, derived);
```

Comment:
I think mq_reset_ids_in_statement () is not needed for dblink node.

Reply:
Ok, I aggre with you.